invoices/update.go

```diff
@@ -191,13 +33,14 @@ func updateInvoice(ctx *invoiceUpdateCtx, inv *channeldb.Invoice) (
 			return nil, ResultReplayToCanceled, nil
 
 		case channeldb.HtlcStateAccepted:
-			return nil, ResultReplayToAccepted, nil
+			return nil, resultReplayToAccepted, nil
```

Comment:
if a literal is returned, e.g. `return nil, 3, nil`, the 3 will be automatically cast into a `ResolutionResult`. the type is not lost because of the function's signature

Reply:
Yes, that part I get. But I was thinking about type checking that `ResolutionResult` later. If it is a plain int, is there still a (specialized) type? Or does golang do boxing/unboxing?

Reply:
Tried it out myself, seems to be boxing indeed